app/services/open_library_service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class OpenLibraryService
+  require 'HTTParty'
+
+  BASE_URI = Rails.application.credentials.config[:api_open_library][:base_uri]
+
+  def initialize(isbn)
+    @isbn = isbn
+  end
+
+  def book_info
+    response = request_to_open_library
+
+    return { status?: :not_found, result: { message: 'Book not found' } } if response.empty?
```

Comment:
Usually for errors we use a different response

Example:
```
{
  "status": 404
  "errors": [
    { "code": "111", "message": "age must be an int" },
    { "code": "112", "message": "email is mandatory" }
  ],
  }
```
  There is no result as this is much cleaner for frontend.
  The code is an internal code to represent the error, you can use the same status code or not add it. But "result" shouldn't be present.

Reply:
https://github.com/Wolox/tech-guides/blob/master/backend/backend-standard-and-best-practices.md

Reply:
The idea of ​​the "result" is to be able to access it as a key and avoid having multiple ifs.

And the response from the endpoint is like this
`{
    "message": "Book not found"
}`

Reply:
But following what you tell me there, the response should be like this:

```
{
   "status":  404,
   "errors": [
         { "message": "Book not found" }
    ]
}
```
But how do I handle those "codes" inside "errors"

Reply:
Those codes are usually codes you create yourself for specific scenarios. For example, YOU define that if X param is missing, then code 001 is the error. if Y param is missing, it's code 002 and so on. I won't bother you with code creation for this scenario as it get's complex to handle, but changing response to errors would be more than enough.
Don't fear IFs. In this scenario it would only prove to need one and it can when it gets too complex it can be handled by middlewares and schema validators.

Reply:
something like that?

![Captura de Pantalla 2021-06-02 a la(s) 20 05 34](https://user-images.githubusercontent.com/53832928/120563047-0fca3c00-c3de-11eb-9b5b-935122ccf0ea.png)